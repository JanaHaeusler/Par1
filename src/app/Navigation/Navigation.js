import styled from 'styled-components/macro'
import { NavLink } from 'react-router-dom'
import { PlusIcon, HomeIcon } from '../Icons/Icons'
import removeLocally from '../../lib/removeLocally'

export default function Navigation() {
  return (
    <NavBar>
      <NavLinkStyled
        exact
        to="/"
        onClick={emptyLocalStorage}
        data-testid="button-home-page"
      >
        <HomeIcon />
      </NavLinkStyled>
      <NavLinkStyled
        to="/create"
        onClick={emptyLocalStorage}
        data-testid="button-form-page"
      >
        <PlusIcon />
      </NavLinkStyled>
    </NavBar>
  )

  function emptyLocalStorage() {
    removeLocally('inputsKeyInfosCreate')
    removeLocally('inputsScoresCreate')
    removeLocally('inputsKeyInfosEdit')
    removeLocally('inputsScoresEdit')
  }
}

const NavBar = styled.nav`
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
`
const NavLinkStyled = styled(NavLink)`
  width: 28px;
  height: 28px;
  fill: var(--light);
  opacity: 40%;

  &.active {
    opacity: 100%;
  }
`
