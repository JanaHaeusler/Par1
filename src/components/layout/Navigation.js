import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import {PlusIcon, HomeIcon} from '../Icons'

Navigation.propTypes = {
    handleClick: PropTypes.func.isRequired
}

export default function Navigation({handleClick}) {
    return (
            <NavBar>
                <NavLinkStyled exact to="/" onClick={handleClick} data-testid="button-home-page">
                    <HomeIcon />
                </NavLinkStyled>
                <NavLinkStyled to="/saveGame" onClick={handleClick} data-testid="button-form-page">
                    <PlusIcon />
                </NavLinkStyled>
            </NavBar>
    )
}

const NavBar = styled.nav`
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 100%;
`
const NavLinkStyled = styled(NavLink)`
    width: 25px;
    fill: var(--text-light);
    opacity: 40%;
    
    &.active {
        opacity: 100%;
    }
`