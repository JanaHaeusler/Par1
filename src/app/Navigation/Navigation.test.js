import Navigation from './Navigation'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'

describe('Navigation', () => {
    
    it('renders correctly', () => {
        const { container } = render(
            <Router>
                <Navigation/>
            </Router> 
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})