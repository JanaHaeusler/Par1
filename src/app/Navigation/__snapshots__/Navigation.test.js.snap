// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c1 {
  width: 25px;
  fill: var(--text-light);
  opacity: 40%;
}

.c1.active {
  opacity: 100%;
}

<nav
  class="c0"
>
  <a
    aria-current="page"
    class="c1 active"
    data-testid="button-home-page"
    href="/"
  >
    <svg>
      home.svg
    </svg>
  </a>
  <a
    class="c1"
    data-testid="button-form-page"
    href="/create"
  >
    <svg>
      plus.svg
    </svg>
  </a>
</nav>
`;
